src/renderers/shaders/ShaderChunk/transmission_pars_fragment.glsl.js

```diff
@@ -184,12 +184,15 @@ export default /* glsl */`
 		// Sample framebuffer to get pixel the refracted ray hits.
 		vec4 transmittedLight = getTransmissionSample( refractionCoords, roughness, ior );
 
-		vec3 attenuatedColor = applyVolumeAttenuation( transmittedLight.rgb, length( transmissionRay ), attenuationColor, attenuationDistance );
+		vec3 transmittance = diffuseColor * volumeAttenuation( length( transmissionRay ), attenuationColor, attenuationDistance );
+		vec3 attenuatedColor = transmittedLight.rgb * transmittance;
 
 		// Get the specular component.
 		vec3 F = EnvironmentBRDF( n, v, specularColor, specularF90, roughness );
 
-		return vec4( ( 1.0 - F ) * attenuatedColor * diffuseColor, transmittedLight.a );
+		float transmittanceFactor = ( transmittance.r + transmittance.g + transmittance.b ) / 3.0;
+
+		return vec4( ( 1.0 - F ) * attenuatedColor, 1.0 - ( 1.0 - transmittedLight.a ) * transmittanceFactor );
```

Comment:
Here's the key to my new hack, which I feel is more defensible than what it replaces. As before, when `transmittedLight.a == 1` (the scene is opaque), no change to the proper PBR computation occurs. As this transmissive material gets lighter and the scene behind gets more transparent, output `alpha` decreases to blend in more of the unknown CSS background color.

Reply:
I am not sure this is needed.

It is best not to hack the physical model if possible.

Reply:
This is addressing a specific opacity bug with your alternative approach: https://github.com/mrdoob/three.js/pull/25881#issuecomment-1520638999

In particular, without this, 20% of the background shows through even if the transmissive material is absorbing all or nearly all of the transmitted light. This hack still has no effect when the final rendered output is opaque.